applications/serial_lte_modem/src/slm_at_fota.c

```diff
@@ -168,27 +169,45 @@ static int do_fota_start(int op, const char *file_uri, int sec_tag,
 		/* UNINIT checker fix, assumed UF_SCHEMA existence */
 		.field_set = 0
 	};
-	char schema[8];
+	char schema[URI_SCHEMA_MAX];
 
 	http_parser_url_init(&parser);
 	ret = http_parser_parse_url(file_uri, strlen(file_uri), 0, &parser);
 	if (ret) {
 		LOG_ERR("Parse URL error");
 		return -EINVAL;
 	}
+
+	/* Schema stores http/https information */
 	memset(schema, 0x00, 8);
 	if (parser.field_set & (1 << UF_SCHEMA)) {
-		strncpy(schema, file_uri + parser.field_data[UF_SCHEMA].off,
-			parser.field_data[UF_SCHEMA].len);
+		if (parser.field_data[UF_SCHEMA].len < URI_SCHEMA_MAX) {
+			strncpy(schema, file_uri + parser.field_data[UF_SCHEMA].off,
+				parser.field_data[UF_SCHEMA].len);
+		} else {
+			LOG_ERR("URL schema too long %d, max %d",
```

Comment:
"URL schema too long 124, max 112" is not very easy to understand. Could this be for example:
"URL schema length 124, exceeds the max length of 112"

Same for others.